linen_examples/seq2seq/train.py

```diff
@@ -223,31 +179,26 @@ def __call__(self, carry, x):
     carry_rng, categorical_rng = jax.random.split(rng, 2)
     if not self.teacher_force:
       x = last_prediction
-    lstm_cell = nn.LSTMCell(name='lstm_cell')
-    projection = nn.Dense(features=self.vocab_size, name='projection')
-    lstm_state, y = lstm_cell(lstm_state, x)
-    logits = projection(y)
-    predicted_tokens = jax.random.categorical(categorical_rng, logits)
-    prediction = onehot(predicted_tokens, self.vocab_size)
+    lstm_state, y = nn.LSTMCell()(lstm_state, x)
+    logits = nn.Dense(features=CTABLE.vocab_size)(y)
+    predicted_token = jax.random.categorical(categorical_rng, logits)
+    prediction = jnp.array(predicted_token == jnp.arange(CTABLE.vocab_size), 
+                           dtype=jnp.float32)
```

Comment:
just curious: why return things in one-hot form instead of just integer ids?  It's probably obvious I'm just dumb about this.

Reply:
I think the whole logic of outputting both logits and predictions is a bit funny. I think it would be easier to only output the logits, and then use them to get the predictions, if necessary. However, in this PR I tried to keep the logic mostly the same except for the automasking, so I think that would be nice as a follow-up. WDYT?

Reply:
Oh sure, I was just curious about it!

Reply:
No problem! I agree with your observation :-)